client/src/main/scala/com.gu.contentapi.client/model/Queries.scala

```diff
@@ -8,9 +8,16 @@ sealed trait ContentApiQuery {
 
   def pathSegment: String
 
-  override def toString = {
-    s"${getClass.getSimpleName}(/$pathSegment${QueryStringParams(parameters)})"
+  override def toString = s"""${getClass.getSimpleName}(${apply("")})"""
+
+  private def url(location: String, parameters: Map[String, String]): String = {
+    require(!location.contains('?'), "must not specify parameters in URL")
+
+    location + QueryStringParams(parameters)
   }
+
+  def apply(targetUrl: String, customParameters: Map[String, String] = Map.empty): String =
```

Comment:
I find `apply` methods for anything but constructors a bit odd, how about `getUrl`?